openstack_controller/tests/test_unit_octavia.py

```diff
@@ -105,3 +105,225 @@ def test_endpoint_up(aggregator, dd_run_check, instance, monkeypatch):
             'project_name:admin',
         ],
     )
+
+
+def test_loadbalancers_metrics(aggregator, dd_run_check, instance, monkeypatch):
+    http = MockHttp("agent-integrations-openstack-octavia")
+    monkeypatch.setattr('requests.get', mock.MagicMock(side_effect=http.get))
+    monkeypatch.setattr('requests.post', mock.MagicMock(side_effect=http.post))
+
+    check = OpenStackControllerCheck('test', {}, [instance])
+    dd_run_check(check)
+
+    base_tags = ['domain_id:default', 'keystone_server:{}'.format(instance["keystone_server_url"])]
+
+    demo_project_tags = base_tags + [
+        'project_id:4762874c945945c38d820cce29fbb66e',
+        'project_name:admin',
+    ]
+
+    demo_loadbalancers = [
+        [
+            'listener_id:de81cbdc-8207-4253-8f21-3eea9870e7a9',
```

Comment:
If there is more than one listener per loadbalancer we should submit these as separate contexts to not have duplicate tags, or make the connection the opposite way